_codegen/go.mod

```diff
@@ -0,0 +1,5 @@
+module github.com/stretchr/testify/_codegen
+
+go 1.13
+
+require github.com/ernesto-jimenez/gogen v0.0.0-20180125220232-d7d4131e6607
```

Comment:
I've added a `go.mod` to `_codegen` so that the `gogen` dependency doesn't show up in the `testify` `go.mod` file when we run go generate. When you run go generate on Go 1.12/1.13 it is discovering the dependency in the `_codegen` directory and adding it to the roots go.mod file. That dependency will become a constraint in any importers dependency graph so it's better to keep the code in here isolated.

Reply:
Should we have `_codegen` as a separate module? If I'm not mistaken, it is custom-written for `testify/assert` and is an integral requirement for generating the usable functions.

Would it make sense to try and keep it simple and only separate it if we have issues from the community about `_codegen` being included?

Reply:
This is the perfect situation for a separate module. As you said it is custom-written for testify, which means we don't want its dependencies to pollute the dependency graph of people importing testify. Codegen is dependent on the `gogen` tool, and without making this its own module anyone importing testify will also have gogen included in their dependency graph.

There's also the problem that the `_codegen` folder starting with an underscore causes it to be ignored by the go mod tooling and so it doesn't show up in `go.mod`, but the moment we do `go run` on the file the `go.mod` gets updated to include it, which causes diffs everytime we run codegen. This is an important part of the change.

Reply:
@glesica Let me know if you'd prefer codegen not be isolated in its own module, and I'll work around that. If we do that I'll need to rename the directory to `codegen` because the underscore causes odd behavior with the go mod commands.

Reply:
Personally, I like the idea of isolating it to keep development dependencies from leaking into applications.

Reply:
Fair enough, it's a good point @leighmcculloch